lightly/loss/macl_loss.py

```diff
@@ -0,0 +1,116 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch import Tensor
+
+
+class MACLLoss(nn.Module):
+    """Implementation of the Model-Aware Contrastive Loss (MACL) from the paper:
+
+    This implementation follows the MACL[0] paper.
+
+    - [0] Model-Aware Contrastive Learning: Towards Escaping the Dilemmas, ICML 2023, https://arxiv.org/abs/2207.07874
+
+    Attributes:
+        t_0: Base temperature. Range: (0, inf)
+        alpha: Scaling factor for controlling how much the temperature changes. Range: [0, 1]
+        A_0: Initial threshold for the alignment magnitude. Range: [0, 1]
+
+    Raises:
+        ValueError:
+            If the initial temperature is less than 1e-8.
+            If the alpha value is not in the range [0, 1].
+            If the A_0 value is not in the range [0, 1].
+
+    Examples:
+        >>> # initialize the loss function
+        >>> loss_fn = MACLLoss()
+        >>>
+        >>> # generate two random transforms of images
+        >>> t0 = transforms(images)
+        >>> t1 = transforms(images)
+        >>>
+        >>> # feed through SimCLR or MoCo model
+        >>> z0 = model(t0)
+        >>> z1 = model(t1)
+        >>>
+        >>> # calculate loss
+        >>> loss = loss_fn(z0, z1)
+    """
+
+    def __init__(self, t_0: float = 0.1, alpha: float = 0.5, A_0: float = 0.0):
+        super().__init__()
+        self.t_0 = t_0
+        self.alpha = alpha
+        self.A_0 = A_0
+        self.eps = 1e-8
+
+        if self.t_0 < self.eps:
+            raise ValueError(
+                "Illegal initial temperature: abs({}) < 1e-8".format(self.t_0)
```

Comment:
We always use f-strings:
```suggestion
                f"Illegal initial temperature: abs({self.t_0}) < 1e-8"
```